KsApi/PKCE.swift

```diff
@@ -0,0 +1,83 @@
+import CryptoKit
+import Foundation
+import Security
+
+enum PKCEError: Error {
+  case UnexpectedRuntimeError
+}
+
+public extension Data {
+  func base64URLEncodedStringWithNoPadding() -> String {
+    var encodedString = self.base64EncodedString()
+
+    // Convert base64 to base64url
+    encodedString = encodedString.replacingOccurrences(of: "+", with: "-")
+    encodedString = encodedString.replacingOccurrences(of: "/", with: "_")
+    // Strip padding
+    encodedString = encodedString.replacingOccurrences(of: "=", with: "")
+
+    return encodedString
+  }
+
+  mutating func fillWithRandomSecureBytes() throws {
+    do {
+      let numBytes = self.count
+      try self.withUnsafeMutableBytes { rawPointer in
+        let pointer = rawPointer.bindMemory(to: UInt8.self)
+        guard let address = pointer.baseAddress else {
+          throw PKCEError.UnexpectedRuntimeError
+        }
+
+        _ = SecRandomCopyBytes(kSecRandomDefault, numBytes, address)
```

Comment:
@ifosli Thanks for reminding me what on earth this method was called - I was looking for it and couldn't find it 😅